sys/net/gnrc/network_layer/sixlowpan/gnrc_sixlowpan.c

```diff
@@ -257,7 +257,7 @@ static void _send(gnrc_pktsnip_t *pkt)
 
     /* IP should not send anything here if it is not a 6LoWPAN interface,
      * so we don't need to check for NULL pointers */
-    if (datagram_size <= iface->max_frag_size) {
+    if (gnrc_pkt_len(pkt2->next) <= iface->max_frag_size) {
```

Comment:
Yes. As commented in https://github.com/RIOT-OS/RIOT/pull/3976/files#diff-e87e12a912a7eb2cbb8a16f4ebeccabaR194: `datagram_size` is the size of the *uncompressed* datagram size. Since that is the size that will be put into the `datagram_size` field of the 6LoWPAN fragmentation header and it can hold only a value of `SIXLOWPAN_FRAG_MAX_LEN` at maximum, the usage in l268 is correct. This change reverses a bug introduced in #3974 (as the description states). Factual it reverses the state and it worked then but it doesn't work now. I currently have no means to test this, so pleasecheck, if fragmentation is working again with this PR. If that's the case I say we can (and must!) merge this one.